M/Matio/build_tarballs.jl

```diff
@@ -0,0 +1,70 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Matio"
+version = v"1.5.23"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/tbeu/matio", "e9e063e08ef2a27fcc22b1e526258fea5a5de329")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+
+cd matio
+
+# trying with cmake
+
+OS=$(uname -s)
+if [ "$OS" = "Darwin" ] ; then
+    LD_EXT="dylib"
+else
+    LD_EXT="so"
+fi
+
+cmake . -DCMAKE_C_COMPILER:FILEPATH=${CC} \
+        -DBUILD_SHARED_LIBS:BOOL=ON \
+        -DMATIO_SHARED:BOOL=ON \
+	    -DMATIO_DEFAULT_FILE_VERSION=7.3 \
+        -DMATIO_MAT73:BOOL=ON \
+        -DCMAKE_INSTALL_PREFIX=${prefix} \
+        -DCMAKE_INSTALL_LIBDIR:PATH=lib \
+        -DMATIO_WITH_HDF5:BOOL=ON \
+        -DMATIO_WITH_ZLIB:BOOL=ON \
+        -DHDF5_ROOT:PATH=${prefix} \
+        -DHDF5_DIR:PATH=${prefix} \
+	    -DZLIB_INCLUDE_DIR:PATH=${prefix}/include \
+        -DZLIB_LIBRARY:FILEPATH=${prefix}/lib/libz.${LD_EXT}
```

Comment:
I'd start by reading the documentation: https://docs.binarybuilder.org/stable/build_tips/#CMake-builds.  In particular, `-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}` is ***not*** optional

Reply:
Thanks @giordano that was very helpful.